src/hooks/screen/useRootChannelListHook.ts

```diff
@@ -29,6 +40,76 @@ const useRootChannelListHook = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
+
+  const checkNotificationPermission = async () => {
+    PushNotification.checkPermissions(
+      (
+        data: PushNotificationPermissions & {
+          authorizationStatus?: number;
+        }
+      ) => {
+        const lastPromptTime = StorageUtils.lastPromptNotification.get();
+        const promptInterval = 48 * 60 * 60 * 1000;
+        const currentTime = new Date().getTime();
+        if (lastPromptTime && currentTime - parseFloat(lastPromptTime) < promptInterval) {
+          return; // Don't show the prompt if the interval has not passed
+        }
+
+        const needsPermission =
+          data.alert === false || data.badge === false || data.sound === false;
+
+        const showAlert = (onPressAction: () => void) =>
+          Alert.alert(
+            "Don't Miss New Messages",
+            `Allow notifications to know 
+when friends send you messages.`,
+            [
+              {text: 'Not now', onPress: () => console.log('Cancel Pressed')},
+              {text: 'Allow', onPress: onPressAction, isPreferred: true}
+            ]
+          );
+
+        switch (data.authorizationStatus) {
+          case PERMISSION_STATUS_ACCEPTED:
+            if (needsPermission) {
+              // THIS CASE IT'S TRIGGERED WHEN USER MANUALLY DISABLED THE NOTIFICATION
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+            } else {
+              requestPermission();
+            }
+            break;
+          case PERMISSION_STATUS_BLOCKED:
+            if (needsPermission) {
+              StorageUtils.lastPromptNotification.set(currentTime.toString());
+              showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
+            }
+            break;
+          case PERMISSION_STATUS_PENDING:
+            StorageUtils.lastPromptNotification.set(currentTime.toString());
+            showAlert(() => requestPermission());
+            break;
+          default:
+            break;
+        }
+      }
+    );
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `checkNotificationPermission` function, there are repeated blocks of code in cases PERMISSION_STATUS_ACCEPTED and PERMISSION_STATUS_BLOCKED (lines 88-92 and 97-100). To follow the DRY principle, you can extract this common logic into a separate function. Also, the console logs on lines 81 and 91 should be replaced with proper error handling or logging mechanisms.

```diff
-          case PERMISSION_STATUS_ACCEPTED:
-            if (needsPermission) {
-              // THIS CASE IT'S TRIGGERED WHEN USER MANUALLY DISABLED THE NOTIFICATION
-              StorageUtils.lastPromptNotification.set(currentTime.toString());
-              showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
-            } else {
-              requestPermission();
-            }
-            break;
-          case PERMISSION_STATUS_BLOCKED:
-            if (needsPermission) {
-              StorageUtils.lastPromptNotification.set(currentTime.toString());
-              showAlert(() => openSettings().catch(() => console.warn('cannot open settings')));
-            }
-            break;

+          case PERMISSION_STATUS_ACCEPTED:
+          case PERMISSION_STATUS_BLOCKED:
+            handlePermissionStatus(needsPermission, currentTime);
+            break;

...

+  const handlePermissionStatus = (needsPermission: boolean, currentTime: number) => {
+    if (needsPermission) {
+      // THIS CASE IT'S TRIGGERED WHEN USER MANUALLY DISABLED THE NOTIFICATION
+      StorageUtils.lastPromptNotification.set(currentTime.toString());
+      showAlert(() => openSettings().catch((error) => {
+        // Replace with your error handling or logging mechanism
+        console.error('Cannot open settings: ', error);
+      }));
+    } else {
+      requestPermission();
+    }
+  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
I just moved the code (from another dev) to hook because the old file is deleted (unnecessary)